<launch>
  <arg name="launch_prefix" default="" /> <!-- set to value="gdbserver localhost:10000" for remote debugging -->
  <arg name="node_namespace" default="apriltag_ros_continuous_node" />
  <arg name="camera_name" default="/image_trim_node" />
  <arg name="camera_frame" default="camera" />
  <arg name="image_topic" default="image_trimmed" />

  <!-- Set parameters -->
  <rosparam command="load" file="$(find apriltag_ros)/config/settings.yaml" ns="$(arg node_namespace)" />
  <rosparam command="load" file="$(find apriltag_ros)/config/tags.yaml" ns="$(arg node_namespace)" />
  
  <node pkg="apriltag_ros" type="apriltag_ros_continuous_node" name="$(arg node_namespace)" clear_params="true" output="screen" launch-prefix="$(arg launch_prefix)" >
    <!-- Remap topics from those used in code to those on the ROS network -->
    <remap from="image_rect" to="$(arg camera_name)/$(arg image_topic)" />
    <remap from="camera_info" to="$(arg camera_name)/camera_info" />

    <param name="camera_frame" type="str" value="$(arg camera_frame)" />
    <param name="publish_tag_detections_image" type="bool" value="true" />      <!-- default: false -->
  </node>
<!--
		<arg name="apriltags3_param" default="[3, 2]"/>
		<rosparam param="apriltags3_param" subst_value="True">"$(arg apriltags3_param)"</rosparam>
-->

<!--
  <group>
    <node name="image_view" pkg="image_view" type="image_view" >
			<param name="autosize" value="true" />
			<remap from="image" to="/usb_cam/image_raw"/>
    </node>
  </group>
-->
  <group>
    <node name="image_view" pkg="image_view" type="image_view" >
			<param name="autosize" value="true" />
			<remap from="image" to="/tag_detections_image"/>
    </node>
<!--
    <node name="image_view2" pkg="image_view" type="image_view" respawn = "false" >
			<param name="autosize" value="true" />
			<remap from="image" to="/usb_cam/image_rect_color"/>
    </node>
		-->
  </group>

	<arg name="manager_name" default="nodelet_manager" />
	<node pkg="nodelet" type="nodelet" name="$(arg manager_name)" args="manager" output="screen"/>
	<node pkg="nodelet" type="nodelet" name="tag_trim_nodelet_node"
		args="load tag_trim/ImageConverter $(arg manager_name)"
		output="screen">
			<remap from="image_topic" to="/usb_cam/image_rect_color"/>
			<remap from="camera_info_topic" to="/usb_cam/camera_info"/>
			<remap from="tag_topic" to="/tag_detections"/>

	</node>


<!--
  <node pkg="tag_trim" type="tag_trim_plugin_nodelet_image_trim_node" name="tag_trim" output="screen" launch-prefix="">
			<remap from="image_topic" to="/usb_cam/image_rect_color"/>
			<remap from="camera_info_topic" to="/usb_cam/camera_info"/>
			<remap from="tag_topic" to="/tag_detections"/>
	</node>
-->
<!--
  <node pkg="apriltag_ros" type="apriltags3.py" name="apriltags3_listener" output="screen"/>
-->
	<rosparam command="load" file="$(find apriltag_ros)/config/apriltags3_config.yaml"/>

</launch>
